drivers/hwmon/max31827.c

```
@@ -543,11 +727,19 @@ static int max31827_init_client(struct max31827_state *st,
 		}
 	}
 
+skip_alarm_pol:
 	if (fwnode_property_present(fwnode, "adi,fault-q")) {
 		ret = fwnode_property_read_u32(fwnode, "adi,fault-q", &data);
 		if (ret)
 			return ret;
 
+		/*
+		 * Fault queue of 6 in MAX31875 has the same mapping as 8 in
+		 * MAX31827, MAX31828 and MAX31829.
+		 */
+		if (type == max31875 && data == 6)
+			data = 8;
```

I actually think this is wrong. We should validate the supported values an error out in case an invalid value is given.